current.go

```diff
@@ -72,12 +73,22 @@ func (w *CurrentWeatherData) SetLang(lang string) error {
 // CurrentByName will provide the current weather with the provided
 // location name.
 func (w *CurrentWeatherData) CurrentByName(location string) error {
-	response, err := http.Get(fmt.Sprintf(fmt.Sprintf(baseURL, "q=%s&units=%s&lang=%s"), location, DataUnits[w.Unit], w.Lang))
-	if err != nil {
-		return err
+	var err error
+	var response *http.Response
+	switch {
+	case strings.Contains(location, " "):
```

Comment:
I saw that you refactored it, thanks.  The whole point is that less code = easier to understand :)